### PR TITLE
Hot Fix: inconsistency of changes since last merge conflicts resolution

### PR DESCRIPTION
**Pull request by DiegoSHS**
## Overview
This pull request fixes some errors in the name of methods used in the subjects service and updated the `Get` route in the subjects controller.

### Details
- **Method:** Updated `find()` to use query params and custom instances of ParseIntPipe to validate the datatype of query params
- **Hot Fix:** Wrong methods names replaced
- **Route:** Added missing delete handler in controller
### Route
Added route handler for http delete requests, this feature was missing, the method for delete was avaliable in the service but no method for delete was present in the controller, it seems like at some point, the route was removed in the last merge conflict resolution.
- `delete()` now avaliable in `subject.controller.ts`

### Hot Fix
Some changes were lost in the last merge conflict resolution
Example of the inconsistency showed in the code base:
```ts
<<<<<<< feature/subjects
    if (await this.foreign.validate())
=======
    if (await this.foreign.validateError())
>>>>>>> dev
```
Notice that `this.foreign.validateError()` doesn't exist and is currently in the **dev branch**

Right now, the merge conflicts are resolved, only need a check to be sure of what changes are being accepted

_**Note:** the second statement was a non-updated change but it was merged together with the other changes_

### CODE REVIEW

**Review by DiegoSHS — COMMENTED** [review]
Auto check, done

**Review by elisa242003 — APPROVED** [review]
Approved, changes implemented correctly

File: src/subject/subject.controller.ts [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+    Get,
+    Param,
+    ParseIntPipe,
+    Post,
+    Put,
+    Query,
 } from '@nestjs/common';
 import { SubjectService } from './subject.service';
 import { CreateSubjectDto } from 'src/models/subject/create-subject.dto';
 import { UpdateSubjectDto } from 'src/models/subject/update-subject.dto';
```
Comment (coby111) [inline src/subject/subject.controller.ts:15]: delete method missing

**Review by coby111 — CHANGES_REQUESTED** [review]
Consider updates

**Review by coby111 — APPROVED** [review]
Changes approved, no comments, everything perfect <3

**Pull request merged**